source/containers/Form/hooks/textReplacement.ts

```diff
@@ -190,136 +205,83 @@ export const replaceText = (
   user: User,
   period?: FormPeriod,
   partner?: PartnerInfo,
-  encryptionPin?: string
-) => {
+  encryptionPin?: string,
+  completionsClarificationMessage?: string
+): string => {
   // This way of doing it might be a bit overkill, but the idea is that this in principle
   // allows for nesting replacement rules and then applying them in order one after the other.
   let res = text ?? "";
   replacementRules.forEach(([template, descriptor]) => {
     res = res.replace(
       template,
-      computeText(descriptor, user, period, partner, encryptionPin)
+      computeText(
+        descriptor,
+        user,
+        period,
+        partner,
+        encryptionPin,
+        completionsClarificationMessage
+      )
     );
   });
   return res;
 };
 
 /**
  * Replaces the markdown as specified by a set of markdown rules and logic that we've defined.
- * @param steps the steps of the form
- * @param user the user object
- * @param period A period to use for dates
  */
 export const replaceMarkdownTextInSteps = (
```

Comment:
You have failing tests and thats because you are changing the `title` property when doing text replacement in `categories` property for questions.`description` should be textreplaced instead of `title`.

You are also missing the `components` property:

```js
components: (question.components ?? []).map((component) => ({
        ...component,
        text: replaceString(component.text),
      })),
```